src/main/java/org/veupathdb/service/sr/response/StreamSequences.java

```diff
@@ -0,0 +1,66 @@
+package org.veupathdb.service.sr.response;
+
+import htsjdk.samtools.reference.IndexedFastaSequenceFile;
+import htsjdk.samtools.util.SequenceUtil;
+import org.veupathdb.service.sr.generated.model.DeflineFormat;
+import org.veupathdb.service.sr.generated.model.Feature;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.function.Consumer;
+
+import htsjdk.tribble.bed.FullBEDFeature;
+import htsjdk.tribble.bed.BEDFeature;
+import htsjdk.tribble.annotation.Strand;
+import htsjdk.samtools.util.Locatable;
+import htsjdk.samtools.util.Interval;
+import java.nio.charset.Charset;
+
+public class StreamSequences {
+
+  private static final Charset CHARSET = Charset.forName("UTF-8");
+
+  private static final byte[] LINE_SEPARATOR = String.valueOf("\n").getBytes(CHARSET);
+
+  public static void write(OutputStream outputStream, IndexedFastaSequenceFile sequences, List<BEDFeature> features, DeflineFormat deflineFormat, int requestedBasesPerLine){
+    int basesPerLine = requestedBasesPerLine > 0 ? requestedBasesPerLine : Integer.MAX_VALUE;
+    try (var buf = new BufferedOutputStream(outputStream)) {
+      for (var feature : features) {
+        var defline = Deflines.deflineForFeature(feature, deflineFormat);
+        var bases = Bases.getBasesForBedFeature(sequences, feature);
```

Comment:
Up to one chromosome in length - so a few megabytes tops - and it would be hard to improve this, because "reverse & complement" currently transposes the whole feature instead of reading backwards.